java/packagescanner.go

```diff
@@ -34,12 +34,14 @@ var (
 )
 
 const DefaultSearchAPI = `https://search.maven.org/solrsearch/select`
+const DefaultRequestTimeout = 10
 
 // ScannerConfig is the struct used to configure a Scanner.
 type ScannerConfig struct {
 	// API is a URL endpoint to a maven-like REST API.
 	// The default is DefaultSearchAPI.
-	API string `yaml:"api" json:"api"`
+	API               string `yaml:"api" json:"api"`
+	APIRequestTimeout int    `yaml:"api_request_timeout" json:"api_request_timeout"`
```

Comment:
```suggestion
	APIRequestTimeout time.Duration `yaml:"api_request_timeout" json:"api_request_timeout"`
```

The timeout should just be a duration.

Reply:
done this to prevent extra memory, time.Duration contains extra params, which are not required.

Reply:
I have done this to prevent add extra memory to struct.

Reply:
That's not a concern for a low-usage struct like this, compared to the documentation and language-level benefit. A time.Duration is an int64 under the hood, so there's no memory saved by making this an int.